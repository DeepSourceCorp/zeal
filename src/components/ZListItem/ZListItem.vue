<template>
  <div class="z-list-item flex space-x-1">
    <z-icon v-if="icon" :icon="icon" :size="iconSize"></z-icon>
    <component :is="as" :href="href" :to="to" :class="[tagSpecificStyles]">
      <slot></slot>
    </component>
  </div>
</template>

<script>
import ZIcon from '@/components/ZIcon/ZIcon.vue'

export default {
  name: 'ZListItem',
  components: {
    ZIcon
  },
  props: {
    as: {
      type: String,
      default: 'div'
    },
    to: {
      type: String
    },
    href: {
      type: String
    },
    icon: {
      type: String
    },
    iconSize: {
      type: String,
      default: 'small'
    }
  },
  data() {
    return {
      styles: {
        a: 'hover:text-juniper transition-all duration-300 ease-in-out'
      }
    }
  },
  computed: {
    tagSpecificStyles() {
      return this.styles[this.as] || ''
    }
  }
}
</script>
