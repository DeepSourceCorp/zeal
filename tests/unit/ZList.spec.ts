import ZList from '../../src/components/ZList'
import ZListItem from '../../src/components/ZListItem'
import { mount } from '@vue/test-utils'

const BasicList = {
  template: `<z-list class="text-vanilla-100 space-y-2">
            <template slot="title">
                <div class="font-medium">Heading</div>
            </template>
            <z-list-item as="a" to="#">First</z-list-item>
            <z-list-item as="a" to="#">Second</z-list-item>
            <z-list-item as="a" to="#">Third</z-list-item>
        </z-list>`,
  components: {
    ZList,
    ZListItem
  }
}

const ListWithTitle = {
  template: `<z-list class="text-vanilla-100 space-y-2" title="Headings">
            <z-list-item as="a" to="#">First</z-list-item>
            <z-list-item as="a" to="#">Second</z-list-item>
            <z-list-item as="a" to="#">Third</z-list-item>
        </z-list>`,
  components: {
    ZList,
    ZListItem
  }
}

describe('ZList', () => {
  it('renders a list with link list items', async () => {
    const wrapper = mount(BasicList)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders a list with title', async () => {
    const wrapper = mount(ListWithTitle)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
