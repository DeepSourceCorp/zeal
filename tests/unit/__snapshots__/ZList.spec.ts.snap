// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZList renders a list with link list items 1`] = `
<div class="z-list flex flex-col text-vanilla-100 space-y-2">
  <div class="font-medium">Heading</div>
  <div class="z-list-item flex space-x-1">
    <!----> <a href="#" class="hover:text-juniper transition-all duration-300 ease-in-out">First</a>
  </div>
  <div class="z-list-item flex space-x-1">
    <!----> <a href="#" class="hover:text-juniper transition-all duration-300 ease-in-out">Second</a>
  </div>
  <div class="z-list-item flex space-x-1">
    <!----> <a href="#" class="hover:text-juniper transition-all duration-300 ease-in-out">Third</a>
  </div>
</div>
`;

exports[`ZList renders a list with title 1`] = `
<div class="z-list flex flex-col text-vanilla-100 space-y-2">
  <div class="font-medium">Headings</div>
  <div class="z-list-item flex space-x-1">
    <!----> <a href="#" class="hover:text-juniper transition-all duration-300 ease-in-out">First</a>
  </div>
  <div class="z-list-item flex space-x-1">
    <!----> <a href="#" class="hover:text-juniper transition-all duration-300 ease-in-out">Second</a>
  </div>
  <div class="z-list-item flex space-x-1">
    <!----> <a href="#" class="hover:text-juniper transition-all duration-300 ease-in-out">Third</a>
  </div>
</div>
`;
